services/whatpulse/whatpulse.service.js

```diff
@@ -0,0 +1,123 @@
+import Joi from 'joi'
+import dayjs from 'dayjs'
+import calendar from 'dayjs/plugin/calendar.js'
+import duration from 'dayjs/plugin/duration.js'
+import relativeTime from 'dayjs/plugin/relativeTime.js'
+import { BaseJsonService } from '../index.js'
+import { metric as formatMetric, ordinalNumber } from '../text-formatters.js'
+dayjs.extend(calendar)
+dayjs.extend(duration)
+dayjs.extend(relativeTime)
+
+const schema = Joi.object({
+  Keys: Joi.alternatives(Joi.string(), Joi.number()).required(),
+  Clicks: Joi.alternatives(Joi.string(), Joi.number()).required(),
+  UptimeSeconds: Joi.alternatives(Joi.string(), Joi.number()).required(),
+  Download: Joi.string().required(),
+  Upload: Joi.string().required(),
+  Ranks: Joi.object({
+    Keys: Joi.string().required(),
+    Clicks: Joi.string().required(),
+    Download: Joi.string().required(),
+    Upload: Joi.string().required(),
+    Uptime: Joi.string().required(),
+  }),
+}).required()
+
+const queryParamSchema = Joi.object({
+  rank: Joi.equal(''),
+}).required()
+
+export default class WhatPulse extends BaseJsonService {
+  static category = 'activity'
+  static route = {
+    base: 'whatpulse',
+    pattern:
+      ':metric(keys|clicks|uptime|download|upload)/:userType(user|team)/:id',
+    queryParamSchema,
+  }
+
+  static examples = [
+    {
+      title: 'WhatPulse user metric',
+      namedParams: { metric: 'keys', userType: 'user', id: '179734' },
+      staticPreview: this.render({
+        metric: 'keys',
+        metricValue: '21G',
+      }),
+    },
+    {
+      title: 'WhatPulse team metric - rank',
+      namedParams: {
+        metric: 'upload',
+        userType: 'team',
+        id: 'dutch power cows',
```

Comment:
Well that's a pretty awesome team id :guitar:

Reply:
I agree :D and they occupy the first rank in all of the categories! 🐮